src/libostree/ostree-deployment.c

```diff
@@ -318,6 +318,8 @@ ostree_deployment_unlocked_state_to_string (OstreeDeploymentUnlockedState state)
       return "hotfix";
     case OSTREE_DEPLOYMENT_UNLOCKED_DEVELOPMENT:
       return "development";
+    case OSTREE_DEPLOYMENT_UNLOCKED_TRANSIENT:
+      return "transient";
```

Comment:
Bikeshed/optional: kinda feels like "development" should be aliased to "transient" instead and this new mode would be "transient-readonly" or something? That would capture better the core difference between the two.

Reply:
Naming is hard for sure...I'd like to avoid renaming the previous one in this PR.  It does capture somewhat though the idea that it should *mostly* be for development - it's mostly about the convenience of just being able to run arbitrary commands to write to the rootfs.

This one though does support that "arbitrary" nature but OTOH I hope its uses will be more constrained in practice.  In particular I'd like to try to build upon this and teach ostree how to *record* in a reliable way (ostree commits?) any changes made in this "underlay".  So that would be more of the admin touch point, rather than just "some transient stuff happened".